src/Expander/Primitives.hs

```diff
@@ -146,7 +148,7 @@ datatype dest outScopesDest stx = do
 
   let info =
         DatatypeInfo
-        { _datatypeArity = fromIntegral arity
+        { _datatypeArgKinds = replicate arity KStar
```

Comment:
the `arity` comes from the number of type parameters given when declaring the datatype, e.g. `(datatype (Either A B) ...` generates a datatype `Either` with arity 2 and thus of kind `* -> * -> *`. That's Haskell's behaviour when `PolyKinds` is off, so I'm thinking of leaving it as-is. One next step would be to allow kind annotations, e.g. `(datatype (StateT (the * s) (the (-> * *) m) (the * a)) ...`.

Reply:
Ideally we'd have kind inference here rather than requiring the annotation. Annotations are nice, but not really in the spirit of the ML-Haskelly language we're building. In any case, `the` should annotate use sites rather than binding sites. What about
`(datatype (StateT s (m (-> * *)) a) ...)` where no annotation means `*` if we absolutely need annotations?

Reply:
In Haskell, I often need to explicitly give kind annotations, because its kind inference defaults to `*` when there is nothing in the type definition which constrains the kind, even when there is some function using that type later in the file which constraints the kind. so it that's also the behaviour we want for kind inference, I do think we need to allow for optional kind annotations.